code/controllers/subsystem/ambient_sound.dm

```diff
@@ -0,0 +1,64 @@
+var/datum/subsystem/ambientsound/SSambience
+//ambient sound subsystem.
+//at the very least it's not a switch right
+
+
+/datum/subsystem/ambientsound
+	name = "Ambient Sound"
+	wait = 5 SECONDS
+	flags = SS_NO_INIT | SS_BACKGROUND | SS_NO_TICK_CHECK
+	priority = SS_PRIORITY_AMBIENCE
+
+
+/datum/subsystem/ambientsound/New()
+	NEW_SS_GLOBAL(SSambience)
+
+
+/datum/subsystem/ambientsound/fire(resumed = FALSE)
+	for (var/client/C in clients)
+		if(C && (C.prefs.toggles & SOUND_AMBIENCE))
+			C.handle_ambience()
+
+/* 
+Ambience system.
+	var/last_ambient_noise //no repeats.
+	var/ambience_buffer // essentially world.time + the length of the ambience sound file. this is to prevent overlap.
+*/
+
+client/proc/handle_ambience()
+	if(!mob)//I have no trust for Byond.
+		return
+
+	var/list/possible_ambience = get_ambience()
+	var/cancel_ambience = TRUE
+	if(last_ambient_noise)
+		for(var/datum/ambience/amb in possible_ambience)
+			if(last_ambient_noise == amb.sound)
+				cancel_ambience = FALSE
+				break
+	if(cancel_ambience)
+		src << sound(null, 0, 0, CHANNEL_AMBIENCE)//I can't think of a sane way to have this be less abrupt. I don't think you can do anything like animating sounds and having a loop for it is gay.
```

Comment:
Am I right in saying that there's no sane way of having the sound fade out rather than cut out?

Reply:
There is a saner way, but that involves finding the sound datum that the user is experiencing.

Look into http://www.byond.com/docs/ref/info.html#/sound for the variables, there is a volume variable associated with the sound datum which you could decrease over a period of time, then when it's near or at silent, you call the sound killer that you have there.

Reply:
The issue is that requires a loop, which I don't really consider a great solution. It should be fine as is, rn the cutoff is typically only at the end of the ambient sound anyway, so it shouldn't be *too* bad. Not worse than it currently is anyway.

Reply:
I wonder if an animate() would work

Reply:
Does animate work with sounds?

Reply:
bah, it doesn't